nucleus/security/core/src/main/java/com/sun/enterprise/security/auth/realm/file/FileRealm.java

```diff
@@ -77,72 +78,88 @@ public final class FileRealm extends IASRealm {
+     * The file provided should always exist. A default (empty) keyfile is installed with the server so this should always
+     * be the case unless the user has manually deleted this file. If this file path provided does not point to an existing
+     * file this constructor will first attempt to create it. If this succeeds the constructor returns normally and an empty
+     * keyfile will have been created; otherwise an exception is thrown.
      *
      * @param keyfile Full path to the keyfile to read for user data.
      * @exception BadRealmException If the configuration parameters identify a corrupt realm.
      * @exception NoSuchRealmException If the configuration parameters specify a realm which doesn't exist.
      *
      */
     public FileRealm(String keyfile) throws BadRealmException, NoSuchRealmException {
-        Properties p = new Properties();
+        Properties properties = new Properties();
+
         // The original keyfile is stored here for use in the init and refresh methods
-        p.setProperty(PARAM_KEYFILE, keyfile);
-        p.setProperty(BaseRealm.JAAS_CONTEXT_PARAM, "ignore");
-        this.init(p);
+        properties.setProperty(PARAM_KEYFILE, keyfile);
+        properties.setProperty(BaseRealm.JAAS_CONTEXT_PARAM, "ignore");
+        this.init(properties);
     }
 
     /*
-     * No arg constructor used by the Realm class when creating realms.
-     * This is followed by a call to the init() method.
+     * No arg constructor used by the Realm class when creating realms. This is followed by a call to the init() method.
      */
     public FileRealm() {
     }
 
+    /**
+     * Returns a short (preferably less than fifteen characters) description of the kind of authentication which is
+     * supported by this realm.
+     *
+     * @return Description of the kind of authentication that is directly supported by this realm.
+     */
+    @Override
+    public String getAuthType() {
+        return AUTH_TYPE;
+    }
+
     /**
      * Return a list of the file names used by all file realms defined for the specified config.
      *
      * @param config the config object
      * @return a list of the file names for all files realms in the config
      */
     public static List<String> getRealmFileNames(Config config) {
-        List<String> files = new ArrayList<>();
-        SecurityService securityService = config.getSecurityService();
-        for (AuthRealm authRealm : securityService.getAuthRealm()) {
+        List<String> realmFileNames = new ArrayList<>();
+
+        for (AuthRealm authRealm : config.getSecurityService().getAuthRealm()) {
             String fileRealmClassName = authRealm.getClassname();
-            // skip it if it's not a file realm
+
+            // Skip it if it's not a file realm
             if (fileRealmClassName == null || !fileRealmClassName.equals(FileRealm.class.getName())) {
                 continue;
             }
+
+            // Skip it if no "file" property
```

Comment:
Comment moved too high, I would put it inside the if block ... or removed it, because it just repeats what we already see.